### PR TITLE
Strimzi version

### PR DESCRIPTION
**Pull request by tombentley**
### Type of change

- Enhancement / new feature

### Description

* Docker images use jar files with version numbers.
* There's some switcheroo with the final names because the shade plugin gets confused it you have the same final name shaded and unshaded when you run mvn twice, which something in the build was doing.
* We have to change the `CMD` (not use array) because docker doesn't do interpolation inside the array. so this means the `CMD` will be passed to `bash -c`
* The approach of using a `release.version` file is stolen from EnMasse. We could use this in future to more easily control the docker tags we publish, to avoid the dreaded unintuitive and essentially semantics-free `latest`. 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,4 @@
+TOPDIR=$(dir $(lastword $(MAKEFILE_LIST)))
```
[the remark this answers is not in the document]
Comment (tombentley) [reply Makefile]: tbh I just copied that from Enmasse. But I can run `make docker_build` from the docker-images subdir and picks up the version frmo the `release.version` file.

File: cluster-controller/pom.xml [hunk 2]
```diff
@@ -76,7 +77,7 @@
                             <goal>shade</goal>
                         </goals>
                         <configuration>
-                            <finalName>cluster-controller</finalName>
+                            <finalName>cluster-controller-${project.version}</finalName>
```
[the remark this answers is not in the document]
Comment (tombentley) [reply cluster-controller/pom.xml]: I think it does. Which deployment tests do you mean (i.e. how can we check?)

File: release.version [hunk 3]
```diff
@@ -0,0 +1 @@
+1.0-SNAPSHOT
```
[the remark this answers is not in the document]
Comment (tombentley) [reply release.version]: OK, done

File: cluster-controller/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (scholzj) [reply cluster-controller/pom.xml]: I mean the `mvn deploy` task. Basically, you should make sure that if you run `mvn install`, it is the fat jar what ends in your local Maven cache and not the small one.

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (scholzj) [reply Makefile]: Ok, then it should work hopefully.

**Comment by scholzj** [comment]
@tombentley One more comment ... we should update the `HACKING.md` file and add to the _Release_ section the new things which should be done after branching the release (such as Bump the Maven version and release.version file).

**Comment by tombentley** [comment]
@scholzj I added a make target for updating versions on `master` following release. If you're happy I'll merge this once CI has passed.

**Pull request merged**